public/main.js

```diff
@@ -1,6 +1,7 @@
 const QUERY_LIMIT = 25
 let dictionary = []
 const DICT_FILEPATH = '/cedict_parsed.csv.gz'
+const definitionCache = {}
```

Comment:
Thanks, I think this is the right direction. Can you leave a TODO here to replace this with a precomputed and more search-efficient data structure (e.g. a trie)?